### PR TITLE
PartialUpdates reporting + additional improvements in error responses

### PR DESCRIPTION
**Pull request by fgalan**
Issue #3499 

- [x] Fix ftest expectations
- [x] Manage correctly the "PartialUpdate" case
- [x] Specific test for this issue (check parent issue and related old PR)
    - [x] POST /v2/op/update cases
    - [x] Other single-entity update operations
- [x] Add entity type to description messages
- [x] Doc (if applicable)
- [x] CNR
- [x] Cleanup comments and FIXME PR marks

### CODE REVIEW

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -3,7 +3,7 @@
 - Fix: correctly detect JSON attribute and metadata value changes in subscription triggering logic (#4211, #4434, #643)
 - Fix: DateTime and geo:json types were not supported in custom notifications using ngsi patching (#4435)
 - Fix: logDeprecate not working correctly (`geo:json` wrongly considered as deprecated)
-- Fix: improve error traces (#4387)
+- Fix: on delete attributes operation the lack of attribute in entity doesn't precules attributes which do exist (this way behaving same way as update attributes operation)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE:6]:
```suggestion
- Fix: improve error traces (#4387)
- Fix: on delete attributes operation the lack of attribute in entity doesn't precules attributes which do exist (this way behaving same way as update attributes operation)
```

Probably the line was accidentality deleted.

File: test/functionalTest/cases/2933_PATCH_offending_attributes/PATCH_offending_attributes.test [hunk 2]
```diff
@@ -77,7 +79,7 @@ echo '03. PATCH /v2/entities/Room1/attrs, to update existing attribute (temperat
 echo '===================================================================================================================================================================='
 payload='{
   "temperature": {
-    "value": 24,
+    "value": 25,
```
Comment (fgalan) [inline test/functionalTest/cases/2933_PATCH_offending_attributes/PATCH_offending_attributes.test:82]: Actually not needed, but better to use a different value than the one in step 02.

NTC (informative)

File: doc/manuals/orion-api.md [hunk 3]
```diff
@@ -2816,15 +2816,25 @@ _**Request headers**_
 _**Response code**_
 
 * Successful operation uses 200 OK
-* Errors use a non-2xx and (optionally) an error payload. See subsection on [Error Responses](#error-responses) for more details.
+* Errors use a non-2xx code and error payload (see next subsection):
```
Comment (mapedraza) [inline doc/manuals/orion-api.md:2819]: Should the link to the error responses section be included? (applies for each API operation)

File: CHANGES_NEXT_RELEASE [hunk 4]
```diff
@@ -4,6 +4,11 @@
 - Fix: DateTime and geo:json types were not supported in custom notifications using ngsi patching (#4435)
 - Fix: logDeprecate not working correctly (`geo:json` wrongly considered as deprecated)
 - Fix: improve error traces (#4387)
+- Fix: on delete attributes operation, the lack of attribute in entity doesn't preclude the deletion of attributes which do exist (this way behaving same way as update attributes operation)
+- Fix: POST /v2/entities/E/attrs?options=append was wrongly creating entities
+- Fix: provide more inforamtive error description in some error responses in update/delete operations
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE:9]:
```suggestion
- Fix: provide more informative error description in some error responses in update/delete operations
```

File: doc/manuals/orion-api.md (hunk 3 above)
Comment (fgalan) [reply doc/manuals/orion-api.md:2819]: The subsection is

```
_**Response payload**_
```

That as far as a I understand, is not link-able.

File: doc/manuals/orion-api.md [hunk 5]
```diff
@@ -2816,15 +2816,26 @@ _**Request headers**_
 _**Response code**_
 
 * Successful operation uses 200 OK
-* Errors use a non-2xx and (optionally) an error payload. See subsection on [Error Responses](#error-responses) for more details.
+* Errors use a non-2xx code and error payload:
+  * 404 Not Found for not found entity (see next subsection)
+  * Check addicional cases in [Error Responses](#error-responses) general documentation
```
[the remark this answers is not in the document]
Comment (fgalan) [reply doc/manuals/orion-api.md]: Fixed (in all places) in [b46c269](https://github.com/telefonicaid/fiware-orion/pull/4447/commits/b46c269bbdd40efb2c25091bae88ab89fc9fb97f)

**Comment by mapedraza** [comment]
LGTM

**Review by mapedraza — APPROVED** [review]

**Comment by fgalan** [comment]
@fisuda this PR does some modifications to English documentation. It would be great if you could sync the Japanese translation, please. Thanks in advance!

**Pull request merged**